### PR TITLE
#256 New Section for PhysicalLinks on component & New Page for

### PR DESCRIPTION
**Pull request by etraisnel2**
PhysicalPort

Added PhysicalPort page
Links towards Physical ports are now navigable

Added Physical Links "Ends" section in component
"Ends" is displayed instead of Source and target for physical links

Change-Id: Ie0fadd9fdf0acb829ab8dc47fda18aeff30b09c7

### CODE REVIEW

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  public PhysicalPortDocGen() {
+    //Here is the constructor
+    StringBuffer stringBuffer = new StringBuffer();
+
+    // add initialisation of the pattern variables (declaration has been already done).
+
+  }
+
+  public String generate(Object argument) throws Exception {
+    final StringBuffer stringBuffer = new StringBuffer();
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:52]: useless final, and why not StringBuilder?

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+    InternalPatternContext ctx = (InternalPatternContext) argument;
+    Map<String, String> queryCtx = null;
+    IQuery.ParameterDescription paramDesc = null;
+    Node.Container currentNode = ctx.getNode();
+
+    List<Object> parameterList = null;
+    //this pattern can only be called by another (i.e. it's not an entry point in execution)
+
+    for (Object parameterParameter : parameterList) {
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:62]: i dont get it, parameterList is null

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+      this.parameter = (org.polarsys.capella.core.data.cs.PhysicalPort) parameterParameter;
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:64]: remove package

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      Map<String, Object> parameterValues = new HashMap<String, Object>();
+      parameterValues.put("parameter", this.parameter);
+      String outputWithCallBack = OutputManager.computeLoopOutput(ictx);
+      String loop = OutputManager.computeLoopOutputWithoutCallback(ictx);
+      ictx.getReporter().loopFinished(loop, outputWithCallBack, ictx, parameterValues);
+    }
+    return null;
+  }
+
+  protected org.polarsys.capella.core.data.cs.PhysicalPort parameter = null;
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:97]: remove package and move parameter to the top of the class

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+  public void set_parameter(org.polarsys.capella.core.data.cs.PhysicalPort object) {
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:99]: same

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    if (ctx.useReporter()) {
+      ctx.getReporter().executionFinished(OutputManager.computeExecutionOutput(ctx), ctx);
+    }
+
+    stringBuffer.append(TEXT_3);
+    stringBuffer.append(TEXT_3);
+    return stringBuffer.toString();
+  }
+
+  public String orchestration(PatternContext ctx) throws Exception {
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:82]: pass directly the type InternalPatternContext

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    InternalPatternContext ictx = (InternalPatternContext) ctx;
+
+    super.orchestration(new SuperOrchestrationContext(ictx));
+
+    if (ictx.useReporter()) {
+      Map<String, Object> parameterValues = new HashMap<String, Object>();
+      parameterValues.put("parameter", this.parameter);
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:89]: Collections.singletonMap(k, v);

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+  protected org.polarsys.capella.core.data.cs.PhysicalPort parameter = null;
+
+  public void set_parameter(org.polarsys.capella.core.data.cs.PhysicalPort object) {
+    this.parameter = object;
+  }
+
+  public Map<String, Object> getParameters() {
+    final Map<String, Object> parameters = new HashMap<String, Object>();
+    parameters.put("parameter", this.parameter);
+    return parameters;
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:106]: return Collections.singletonMap(k, v);

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+import org.polarsys.capella.docgen.util.CapellaComponentServices;
+import org.polarsys.capella.docgen.util.StringUtil;
+import org.polarsys.capella.docgen.util.pattern.helper.BlockHelper;
+import org.polarsys.kitalpha.doc.gen.business.core.util.EscapeChars;
+import org.polarsys.capella.core.data.oa.Entity;
+import org.polarsys.capella.docgen.util.pattern.helper.CapellaComponentPortHelper;
+import org.polarsys.capella.core.data.cs.PhysicalPort;
+import org.polarsys.capella.core.data.cs.PhysicalLink;
+
+public class PhysicalPortDocGen extends org.polarsys.capella.docgen.foundations.NamedElementDocGen {
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:21]: add annotation override to overridden methods

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+    if (providedInterfaces.size() > 0) {
+
+      stringBuffer.append(TEXT_2);
+      stringBuffer.append(TEXT_3);
+      stringBuffer.append(StringUtil.mapToHTMLTable(providedInterfaces, "Interface", "Exchange Items"));
+
+    }
+    Map<String, String> requiredInterfaces = CapellaComponentPortHelper.getRequiredInterfaces(parameter, projectName,
+        outputFolder);
+    if (requiredInterfaces.size() > 0) {
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:128]: !map.isEmpty
same for below ones

File: docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaComponentServices.java [hunk 11]
```diff
@@ -150,6 +165,80 @@ public static String componentExchangeToTableLine(Component currentComponent, Co
...
+				EObject container = end.eContainer();
+				buffer.append(CapellaServices.LI_OPEN);
+				buffer.append(CapellaServices.getImageLinkFromElement(container, projectName, outputFolder));
+				buffer.append(" ");
+				buffer.append(CapellaServices.getHyperlinkFromElement(container));
+				buffer.append(CapellaServices.LI_CLOSE);
+			}
+			buffer.append(CapellaServices.UL_CLOSE);
+		}
+		buffer.append(CapellaServices.TD_CLOSE);
```
Comment (mchoubby) [inline docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaComponentServices.java:192]: code duplicated 4 times 
should be in a separate method

Comment (etraisnel2) [reply docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaComponentServices.java]: This particular comment was dealt with
All other comments are about gen code, which is automatically generated when building after the fcore was updated

**Review by mchoubby — APPROVED** [review]

File: docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java (hunk 10 above)
Comment (pdulth) [reply docgenhtml/plugins/org.polarsys.capella.docgen/generated/org/polarsys/capella/docgen/content/PhysicalPortDocGen.java:128]: This class is generated automatically

**Review by pdulth — APPROVED** [review]

**Pull request merged**